app/src/lib/stores/app-store.ts

```diff
@@ -2918,11 +2918,10 @@ export class AppStore {
     const branch = tip.branch
     const aheadBehind = state.aheadBehind
 
-    if (!aheadBehind) {
+    if (aheadBehind == null) {
       this._showPopup({
-        type: PopupType.PushBranchCommits,
+        type: PopupType.PublishRepository,
```

Comment:
The other reason I don't think this is the right fix is that the "Create Pull Request" button should only be enabled for a GitHub or Enterprise repository - which assumes that it has a remote setup. Publishing a repository should only be necessary if that remote doesn't exist...